datafusion/proto/tests/cases/roundtrip_logical_plan.rs

```diff
@@ -602,32 +551,6 @@ fn round_trip_scalar_values() {
             i64::MAX,
         ))),
         ScalarValue::IntervalMonthDayNano(None),
-        ScalarValue::new_list(
```

Comment:
The reason I wrote another is because round_trip_scalar_values is too large and I just want to test the list only, so I wrote it in another test.

Reply:
> It seems like it would be easier to understand the ListArray coverage if we kept the test structure as is 🤔
Nice suggestion, it looks really nice now